hs_file_types/models/netcdf.py

```diff
@@ -422,53 +423,24 @@ def set_file_type(cls, resource, user, file_id=None, folder_path=None):
-            create_new_folder = cls._check_create_aggregation_folder(
-                selected_res_file=res_file, selected_folder=folder_path,
-                aggregation_file_count=1)
 
             with transaction.atomic():
                 # create a netcdf logical file object to be associated with
                 # resource files
                 dataset_title = res_dublin_core_meta.get('title', nc_file_name)
                 logical_file = cls.initialize(dataset_title, resource)
-
+                # create logical file record in DB
+                logical_file.save()
                 try:
                     if folder_path is None:
                         # we are here means aggregation is being created by selecting a file
-
-                        # create a folder for the netcdf file type using the base file
-                        # name as the name for the new folder if the file is not already in a folder
-                        if create_new_folder:
-                            upload_folder = cls._create_aggregation_folder(resource, file_folder,
-                                                                           nc_file_name)
-                            aggregation_folder_created = True
-                            log.info("NetCDF Aggregation creation - folder created:{}".format(
-                                upload_folder))
-                        else:
-                            # selected nc file is already in a folder
-                            upload_folder = file_folder
-
-                        # create logical file record in DB
-                        logical_file.save()
-                        if aggregation_folder_created:
-                            # copy the nc file to the new aggregation folder and make it part
-                            # of the logical file
-                            tgt_folder = upload_folder
-                            files_to_copy = [res_file]
-                            logical_file.copy_resource_files(resource, files_to_copy,
-                                                             tgt_folder)
-                            res_files_to_delete.append(res_file)
-                        else:
-                            # make the selected nc file as part of the aggregation/file type
-                            logical_file.add_resource_file(res_file)
-
+                        upload_folder = file_folder
                     else:
-                        # logical file record gets created in DB
-                        logical_file.save()
                         # folder has been selected to create aggregation
                         upload_folder = folder_path
-                        # make the .nc file part of the aggregation
-                        logical_file.add_resource_file(res_file)
+
+                    # make the .nc file part of the aggregation
+                    logical_file.add_resource_file(res_file)
```

Comment:
Seems like there is a lot of duplicated code in set_file_type across the LogicalFile types.  The only unique implementation in these methods occurs when parsing the metadata.  This method could be implemented in AbstractLogicalFile and then each implementation would only have to implement the metadata parsing.